engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsSameObjectNode.java

```diff
@@ -50,24 +50,12 @@ boolean isSameType(Type typeLeft, Type typeRight) {
    *
    * @return True if the qualified names of the meta objects are same.
    */
-  @Specialization(
-      guards = {
-        "!interop.isNull(metaLeft)",
-        "interop.isMetaObject(metaLeft)",
-        "!interop.isNull(metaRight)",
-        "interop.isMetaObject(metaRight)"
-      })
+  @Specialization(guards = {"interop.isMetaObject(metaLeft)", "interop.isMetaObject(metaRight)"})
   boolean isSameMetaObjects(
       Object metaLeft,
       Object metaRight,
       @Shared("interop") @CachedLibrary(limit = "2") InteropLibrary interop) {
-    try {
-      Object metaLeftName = interop.getMetaQualifiedName(metaLeft);
-      Object metaRightName = interop.getMetaQualifiedName(metaRight);
-      return isIdenticalObjects(metaLeftName, metaRightName, interop);
-    } catch (UnsupportedMessageException e) {
-      throw EnsoContext.get(this).raiseAssertionPanic(this, null, e);
-    }
+    return isIdenticalObjects(metaLeft, metaRight, interop);
```

Comment:
Does this specialization still even make sense? Now, you just delegate to a fallback. I would suggest you to remove this specialization and just leave the code in the fallback. See if tests will be passing. That will be simplification.